test/iaas.GcpClient.spec.js

```diff
@@ -0,0 +1,538 @@
+      });
+    });
+
+    describe('#GcpCompute', function () {
+      it('should form an object with credentials and projectId with compute base url', function () {
+        const responseGcpComputeObject = GcpClient.createComputeClient(settings);
+        const baseUrl = 'https://www.googleapis.com/compute/v1';
+
+        expect(responseGcpComputeObject.baseUrl).to.equal(baseUrl);
+        expect(responseGcpComputeObject.projectId).to.equal(settings.projectId);
+        expect(responseGcpComputeObject.authClient.config.credentials).to.eql(settings.credentials);
+      });
+    });
+
+    describe('#BlobOperations', function () {
+      before(function () {
+        sandbox.stub(GcpStorage.prototype, 'bucket').withArgs(settings.container).returns(bucketStub);
+        sandbox.stub(utils, 'streamToPromise', utilsStreamToPromiseStub);
+      });
+      after(function () {
+        sandbox.restore();
+      });
+
+      it('container properties should be retrived successfully', function () {
+        return client.getContainer()
+          .then(result => {
+            expect(result[1].name).to.equal(settings.container);
+            expect(result[1].id).to.equal(settings.container);
+            expect(result[1].location).to.equal(region);
+            expect(result[1].storageClass).to.equal(storageClass);
+            expect(result[1].selfLink).to.equal(storageUrl + bucketUrl + '/' + settings.container);
+          });
+      });
+
+      it('list of files/blobs should be returned', function () {
+        const options = {
+          prefix: 'blob'
+        };
+        const expectedResponse = [{
+            name: 'blob1.txt'
+          },
+          {
+            name: 'blob2.txt'
+          }
+        ];
+        return client.list(options)
+          .then(result => {
+            expect(result).to.be.an('array');
+            expect(result).to.have.lengthOf(2);
+            expect(result).to.be.eql(expectedResponse);
+          });
+      });
+
+      it('file/blob deletion should be successful', function () {
+        return client.remove(validBlobName)
+          .then(result => expect(result[0]).to.be.undefined);
+      });
+
+      it('file/blob deletion should be throw NotFound exception', function () {
+        return client.remove(notFoundBlobName)
+          .catch(err => expect(err).to.be.instanceof(NotFound));
+      });
+
+      it('file/blob deletion should be throw Forbidden exception', function () {
+        return client.remove('blob4.txt')
+          .catch(err => expect(err).to.be.instanceof(Forbidden));
+      });
+
+      it('file/blob upload should be successful', function () {
+        var uploadJsonPromise = client.uploadJson(validBlobName, jsonContent);
+        return Promise.delay(CONNECTION_WAIT_SIMULATED_DELAY)
+          .then(() => {
+            if (streamEventHandlers.finish && _.isFunction(streamEventHandlers.finish)) {
```

Comment:
it would be a good idea to reset the object `streamEventHandlers` to empty afterEach test run. Else it will carry refereces from previous test.

Reply:
Sure, added below: 
```
      afterEach(function () {
        streamEventHandlers = {};
      });
```
Also added the same for `deleteSnapshotEventHandlers` in #ComputeOperations.